zsh-copilot.plugin.zsh

```diff
@@ -82,44 +70,84 @@ function _suggest_ai() {
                 }
             ]
         }"
+    
     local response=$(curl "https://${OPENAI_API_URL}/v1/chat/completions" \
         --silent \
         -H "Content-Type: application/json" \
         -H "Authorization: Bearer $OPENAI_API_KEY" \
         -d $data)
-    local message=$(echo "$response" | jq -r '.choices[0].message.content')
 
-    # zle -U "$suggestion"
+    local message=$(echo "$response" | jq -r '.choices[0].message.content')
 
     local first_char=${message:0:1}
     local suggestion=${message:1:${#message}}
-    
+
     if [[ "$ZSH_COPILOT_DEBUG" == 'true' ]]; then
         touch /tmp/zsh-copilot.log
         echo "$(date);INPUT:$input;RESPONSE:$response;FIRST_CHAR:$first_char;SUGGESTION:$suggestion:DATA:$data" >> /tmp/zsh-copilot.log
     fi
 
     if [[ "$first_char" == '=' ]]; then
-        # Reset user input
         BUFFER=""
         CURSOR=0
+        zle -U "$suggestion"
+    elif [[ "$first_char" == '+' ]]; then
+        _zsh_autosuggest_suggest "$suggestion"
+    fi
+}
+
+# Function for activation via Ollama (triggered by ^e)
+function _suggest_ai_ollama() {
+    if [[ "$ZSH_COPILOT_SEND_CONTEXT" == 'true' ]]; then
+        local PROMPT="$SYSTEM_PROMPT 
+            Context: You are user $(whoami) with id $(id) in directory $(pwd). 
+            Your shell is $(echo $SHELL) and your terminal is $(echo $TERM) running on $(uname -a).
+            $SYSTEM"
+    fi
 
+    local input=$(echo "${BUFFER:0:$CURSOR}" | tr '\n' ';')
+    input=$(echo "$input" | sed 's/"/\\"/g')
+
+    _zsh_autosuggest_clear
+    zle -R "Thinking (Ollama)..."
+
+    PROMPT=$(echo "$PROMPT" | tr -d '\n')
+
+    # Call to Ollama using the llama3 model
+    local response=$(echo "$PROMPT $input" | ollama run llama3)
```

Comment:
Maybe you could call the [ollama API](https://github.com/ollama/ollama/blob/main/docs/api.md#generate-a-completion) with HTTP, so that you can configure the URL to be a remote server if you want to run ollama on another computer/server. What do you think? :)
(Be sure to use `stream=false` to ensure to get the whole respone)

Reply:
Why not in the future, to be honest, I haven't thought about this type of Ollama implementation :)